src/linter.js

```diff
@@ -629,4 +634,25 @@ export default class Linter {
       });
   }
 
+  _markBadwordUsage(filename, fileData) {
+    if (fileData) {
+      const badwordsRe = new RegExp(badwords.en.join('|'), 'gi');
+
+      const words = fileData.split(' ');
+
+      for (let word of words) {
+        // Filter out none a-Z characters to simplify matching with our
+        // badword list.
+        word = word.toLowerCase().replace(/[^a-zA-Z]/, '');
+
+        if (word.match(badwordsRe)) {
```

Comment:
I am a bit worried about the performance impacts of these broad regular expressions. Especially line 639 could be quite slow. Have we measured this with some sample set or real add-ons?